frappe/auth.py

```diff
@@ -123,7 +124,18 @@ def login(self):
 		# clear cache
 		frappe.clear_cache(user = frappe.form_dict.get('usr'))
 		user, pwd = get_cached_user_pass()
-		self.authenticate(user=user, pwd=pwd)
+		try:
```

Comment:
Very bad idea to catch unnamed exceptions like this.

make this explicit, if ldap is enabled, then login via ldap first, then regular